swiftsimio/visualisation/slice_backends/nearest_neighbors.py

```diff
@@ -0,0 +1,236 @@
+from numpy import float64, float32, ndarray, linspace, array, stack, meshgrid, cbrt
+
+from swiftsimio import SWIFTDataset, cosmo_array
+from swiftsimio.optional_packages import KDTree, TREE_AVAILABLE
+from swiftsimio.visualisation.slice_backends.sph import get_hsml as sph_get_hsml
```

Comment:
My final comment is that we should probably move get_hsml to a higher level? As you will likely want to do projections too.

You could add this at the swiftsimio.visualisation level?

Reply:
Yes, good idea